test/libgit2.jl

```diff
@@ -1825,7 +1825,6 @@ mktempdir() do dir
                 # User provides an empty private key which triggers a re-prompt
                 challenges = [
                     "Private key location for 'git@github.com':" => "\n",
-                    "Public key location for 'git@github.com' [.pub]:" => "\n",
```

Comment:
I understand that this test doesn't make sense now that we're only checking for the public key when the private key is provided, but don't we still need to check the public key prompting when a valid private key is provided?

Reply:
There is one test I can add for public keys at this point. There's one round left of refactoring that will allow me to add a few more tests which currently cause the `credential_loop` to infinite loop.